### ISSUE TITLE
Migrate to pnpm

### ISSUE THREAD

**Issue by sublimator** [interledger/web-monetization-projects#3733]

**Comment by sublimator** [comment]
```
  nodeId: 'registry.npmjs.org/convert-source-map/1.9.0'
}
nodeJ {
  alias: 'debug',
  node: undefined,
  nodeId: 'link:patch:debug@npm%253A4.3.4#~/.yarn/patches/debug-npm-4.3.4-4513954577.patch'
 ERROR  Cannot read properties of undefined (reading 'depth')

pnpm: Cannot read properties of undefined (reading 'depth')
    at toPkgByName (/Users/nicholasdudfield/Library/pnpm/global/5/.pnpm/pnpm@8.2.0/node_modules/pnpm/dist/pnpm.cjs:183421:23)
    at resolvePeersOfNode (/Users/nicholasdudfield/Library/pnpm/global/5/.pnpm/pnpm@8.2.0/node_modules/pnpm/dist/pnpm.cjs:
```

Need to first deal with the `yarn patch` resolutions

**Comment by sublimator** [comment]
"resolutions": {
    "@types/react": "17.0.56",
    "@types/serve-static": "1.15.1",
    "inversify": "patch:inversify@npm:6.0.1#.yarn/patches/inversify-npm-6.0.1-39ef6784da",
    "@wext/shipit": "patch:@wext/shipit@npm:0.2.1#.yarn/patches/@wext-shipit-npm-0.2.1-0aefb3f515",
    "debug": "patch:debug@npm%3A4.3.4#~/.yarn/patches/debug-npm-4.3.4-4513954577.patch",
    "ilp-plugin-btp@npm:^1.3.5": "patch:ilp-plugin-btp@npm%3A1.5.0#~/.yarn/patches/ilp-plugin-btp-npm-1.5.0-f669b16b57.patch"
  },

**Comment by sublimator** [comment]
Actually, going by experience with interledgerjs/interledgerjs, it seems pnpm could be overly liberal wrt: shared packages. 
It would be nice if devDependencies were inherited from the root workspace but other than that each package should be isolated.

**Comment by sublimator** [comment]
Actually, it seems that pnpm has native support for patching
https://pnpm.io/cli/patch

**Comment by sublimator** [comment]
Closed in #3754

**Comment by sublimator** [comment]
For the record some regrets. As much of a hassle yarn2/pnp can be at times, the day-to-day speeeeeeed is so much better there's an argument to be made it's "worth it"
